src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -47,6 +47,12 @@ export const DiscoverTopNav = ({
   const dataView = useInternalStateSelector((state) => state.dataView!);
   const savedDataViews = useInternalStateSelector((state) => state.savedDataViews);
   const savedSearch = useSavedSearchInitial();
+  const [dataViewList, setDataViewList] = useState([dataView]);
+  useEffect(() => {
+    if (!dataViewList.includes(dataView)) {
+      setDataViewList([dataView, ...dataViewList]);
+    }
+  }, [dataView, dataViewList, setDataViewList]);
```

Comment:
Would be better to remove dataViewList from deps:

```suggestion
  const isDataViewMissing = !dataViewList.includes(dataView);
  
  useEffect(() => {
    if (isDataViewMissing) {
      setDataViewList(currentDataViewList => [dataView, ...currentDataViewList.filter(dw => dw.id !== dataView.id)]);
    }
  }, [dataView, isDataViewMissing, setDataViewList]);
```

Reply:
@jughosta makes sense, thx!